core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -163,39 +174,24 @@ private static Properties createDefaultProps() {
    *
    * @param properties The source {@link Properties} to be merged
    */
-  public static void merge(Map<?, ?> properties) {
+  public static void merge(Map<?, ?> properties, Source source) {
     if (properties != null) {
-      // merge the system properties
+      // merge the properties
       for (Map.Entry<?, ?> entry : properties.entrySet()) {
         String key = entry.getKey().toString().trim();
         String value = entry.getValue().toString().trim();
         if (PropertyKey.isValid(key)) {
+          PropertyKey propertyKey = PropertyKey.fromString(key);
+          if (source == Source.SITE_PROPERTY && propertyKey.isIgnoredSiteProperty()) {
+            LOG.warn("{} is not accepted in alluxio-site.properties, "
+                    + "and must be specified as a JVM property. "
+                    + "If no JVM property is present, Alluxio will use default value '{}'.",
+                key, propertyKey.getDefaultValue());
+            continue;
```

Comment:
We may want to crash here.

Reply:
move this logic to validation and will crash there to be consistent.